### PR TITLE
added enzyme and tested Link/Provider components

### PR DESCRIPTION
**Pull request by JForsaken**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -40,22 +40,24 @@
...
     "webpack-node-externals": "^1.6.0"
   },
-  "peerDependencies": {
-    "history": "^4.6.3"
-  },
   "dependencies": {
     "lru-memoize": "^1.0.2",
-    "query-string": "^4.3.4"
+    "query-string": "^4.3.4",
+    "history": "^4.6.3"
```
Comment (JForsaken) [inline package.json]: @tomzmtl il fallait que je mette `history` dans les dependencies sinon ca plantait.. c'est normal?

Comment (tomzmtl) [inline package.json]: Shouldn't. `history` may be needed to run the tests but shouldn't be added to the bundle.
Try installing the peerDeps instead.

File: package.json [hunk 2]
```diff
@@ -11,9 +11,9 @@
     "dev": "webpack --watch",
     "lint": "eslint ./src --ext .js,.jsx",
     "prepublish": "npm run build",
-    "test": "NODE_ENV=test mocha --opts mocha.opts || true",
+    "test": "NODE_ENV=test mocha --require src/tests/setup.js --opts mocha.opts || true",
```
Comment (tomzmtl) [inline package.json]: The `--require` line should be put in `mocha.opts` and `mocha.travis.opts`.

File: package.json [hunk 3]
```diff
@@ -40,21 +40,26 @@
...
     "redux": "^3.7.1",
+    "sinon": "^2.3.8",
     "webpack": "^3.0.0",
     "webpack-node-externals": "^1.6.0"
   },
   "peerDependencies": {
     "history": "^4.6.3"
   },
   "dependencies": {
+    "jsdom": "^11.1.0",
```
Comment (tomzmtl) [inline package.json]: This should be in `devdependencies`.

File: src/tests/.eslintrc.js [hunk 4]
```diff
@@ -2,4 +2,7 @@ module.exports = {
   rules: {
     'max-len': 'off',
   },
+  globals: {
+    global: true,
```
Comment (tomzmtl) [inline src/tests/.eslintrc.js]: This should be

```
{
  env: {
    node: true,
  }
}
```

**Pull request merged**